File: tldr-api/src/main/kotlin/run/cd80/tldr/api/auth/DefaultSignInUser.java
package run.cd80.tldr.api.auth;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.util.Assert;

import java.util.Collection;
import java.util.Collections;
import java.util.List;
import java.util.Map;

public class DefaultSignInUser implements SignInUser {

    private final String username;

    public DefaultSignInUser(final String username) {
        Assert.hasText(username, "username cannot be empty");

        this.username = username;
    }

    @Override
    public String getName() {
        return this.username;
    }

    @Override
    public Map<String, Object> getAttributes() {
        return Collections.emptyMap();
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return List.of(new SimpleGrantedAuthority("ROLE_USER"));
    }
}

File: tldr-api/src/main/kotlin/run/cd80/tldr/api/auth/DefaultSignInUser.kt
package run.cd80.tldr.api.auth

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.authority.SimpleGrantedAuthority

class DefaultSignInUser(private val username: String) : SignInUser {

    override fun getName(): String =
        username

    override fun getAttributes(): MutableMap<String, Any> =
        mutableMapOf()

    override fun getAuthorities(): MutableCollection<out GrantedAuthority> =
        mutableListOf(SimpleGrantedAuthority("ROLE_USER"))
}
